src_py/sprite.py

```diff
@@ -245,6 +249,35 @@ def layer(self, value):
                 "instead."
             )
 
+    def handle_event(self,event):
+        """method to handle events
+
+        Sprite.handle_event(event):
+
+        The default implementation of this method does nothing; it's just a
+        convenient "hook" that you can override. This method is called by
+        Group.handle_event(event) with the event you called it with only if
+        it's type is in listening_events.
+
+        Note that if you change listening_events, the changes will not be 
+        acknoleged untill you call Sprite.update_listening_events()
+
+        There is no need to use this method if not using the convenience
+        method by the same name in the Group class.
+        """
+
+    def update_listening_events(self):
+        for event_type in self.listening_events-self._old_listening_event:
+            #The previous line only interates over new event types
+            for group in self.__g:
+                group._register_event(event_type,self)
+        for event_type in self._old_listening_event-self.listening_events:
+            #The previous line only interates over old event types
```

Comment:
Btw, the comments should have a space between the hastag and the actual message content